### PR TITLE
Add `name` and `geo` to `User`

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

- Add `name` property to `User`
- Add `Geo` class
- Add `geo` property to `User`

## :bulb: Motivation and Context

Relates to https://github.com/getsentry/sentry-dart/issues/1065

## :green_heart: How did you test it?

Added unit tests

## :pencil: Checklist

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

## :crystal_ball: Next steps

Implement on cocoa and use in hybrid SDKs

### CODE REVIEW

**Comment by philipphofmann** [comment]
@denrase, did you tag me for review for a specific reason? If not, Roman or Markus could also have a look at this PR.

File: sentry/src/main/java/io/sentry/protocol/Geo.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    }
+    writer.endObject();
+  }
+
+  public static final class Deserializer implements JsonDeserializer<Geo> {
+
+    @Override
+    public Geo deserialize(JsonObjectReader reader, ILogger logger) throws Exception {
+      reader.beginObject();
+      Geo geo = new Geo();
```
Comment (markushi) [inline sentry/src/main/java/io/sentry/protocol/Geo.java:138]: could be marked as final

File: sentry/src/main/java/io/sentry/protocol/Geo.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class Geo implements JsonUnknown, JsonSerializable {
+
+  /** Human readable city name. * */
+  @Nullable private String city;
+
+  /** Two-letter country code (ISO 3166-1 alpha-2). * */
+  @Nullable private String countryCode;
```
Comment (markushi) [inline sentry/src/main/java/io/sentry/protocol/Geo.java:22]: @romtsn should we enforce this format somehow? Our unified docs mention this as well: https://develop.sentry.dev/sdk/event-payloads/user/#geographical-location

File: sentry/src/main/java/io/sentry/protocol/Geo.java (hunk 1 above)
Comment (markushi) [inline sentry/src/main/java/io/sentry/protocol/Geo.java:138]:
```suggestion
      final Geo geo = new Geo();
```

We usually like to mark things as much `final` as possible, could you add it here and to other places as well?

File: sentry/src/test/resources/json/user.json [hunk 3]
```diff
@@ -3,6 +3,12 @@
     "id": "efb2084b-1871-4b59-8897-b4bd9f196a01",
     "username": "60c05dff-7140-4d94-9a61-c9cdd9ca9b96",
     "ip_address": "51d22b77-f663-4dbe-8103-8b749d1d9a48",
+    "name": "c8c60762-b1cf-11ed-afa1-0242ac120002",
+    "geo": {
+        "city": "0e6ed0b0-b1c5-11ed-afa1-0242ac120002",
+        "country_code": "16717632-b1c5-11ed-afa1-0242ac120002",
```
Comment (markushi) [inline sentry/src/test/resources/json/user.json:9]: `low`: maybe use a better value here
```suggestion
        "country_code": "JP",
```

File: sentry/src/test/java/io/sentry/protocol/UserSerializationTest.kt [hunk 4]
```diff
@@ -21,6 +21,12 @@ class UserSerializationTest {
             id = "efb2084b-1871-4b59-8897-b4bd9f196a01"
             username = "60c05dff-7140-4d94-9a61-c9cdd9ca9b96"
             ipAddress = "51d22b77-f663-4dbe-8103-8b749d1d9a48"
+            name = "c8c60762-b1cf-11ed-afa1-0242ac120002"
+            geo = Geo().apply {
+                city = "0e6ed0b0-b1c5-11ed-afa1-0242ac120002"
+                countryCode = "16717632-b1c5-11ed-afa1-0242ac120002"
```
Comment (markushi) [inline sentry/src/test/java/io/sentry/protocol/UserSerializationTest.kt:27]:
```suggestion
                countryCode = "JP"
```

**Review by markushi — CHANGES_REQUESTED** [review]
Looking great, thanks for contributing - please have a look at my comments 🚀

**Review by markushi — APPROVED** [review]
Nice one, LGTM!

File: sentry/src/main/java/io/sentry/protocol/Geo.java (hunk 2 above)
Comment (romtsn) [reply sentry/src/main/java/io/sentry/protocol/Geo.java:22]: I think it's fine, we have a few places already where we are expected to send data in certain format, but do not enforce it anyhow (e.g. `release`), we can think about it later I guess

**Review by romtsn — APPROVED** [review]
@denrase wanna go ahead and merge it?

File: sentry-android-ndk/sentry-native
[the remark this answers is not in the document]
Comment (denrase) [reply sentry-android-ndk/sentry-native:1]: will do and merge, thank you!

**Comment by romtsn** [comment]
> @romtsn A UI running on CI test seems to be flaky. AFAIK this should not be connected to this PR. Can we merge or is there something to do first?

merge it!

**Comment by denrase** [comment]
Thx @markushi & @romtsn for the reviews!

**Pull request merged**